### PR TITLE
portals page

### PR DESCRIPTION
**Pull request by mayankverr**

### CODE REVIEW

File: content/docs/portals.md [hunk 1]
```diff
@@ -4,21 +4,21 @@ title: Portals
 permalink: docs/portals.html
 ---
 
-Portals provide a first-class way to render children into a DOM node that exists outside the DOM hierarchy of the parent component.
+Portals चिल्ड्रन को एक ऐसे DOM नोड में रेंडर करने का तरीका प्रदान करते हैं जो पैरेंट कौम्पोनॅन्ट के DOM हायरार्की के बाहर मौजूद हो।
```
Comment (arshadkazmi42) [inline content/docs/portals.md:7]: `Portals` => `पोर्टल्स`

File: content/docs/portals.md [hunk 2]
```diff
@@ -27,34 +27,34 @@ render() {
...
 ```
 
-A typical use case for portals is when a parent component has an `overflow: hidden` or `z-index` style, but you need the child to visually "break out" of its container. For example, dialogs, hovercards, and tooltips.
+Portals का एक विशिष्ट उपयोग का मामला है जब एक पैरेंट कौम्पोनॅन्ट में `overflow: hidden` या `z-index` स्टाइल होता है, लेकिन आपको चाइल्ड को नेत्रहीन उसके कंटेनर से "बाहर" करने की आवश्यकता है। जैसे डॉयलोग्स, होवरकार्ड और टूलटिप्स।
 
-> Note:
+> टिप्पणी:
 >
-> When working with portals, remember that [managing keyboard focus](/docs/accessibility.html#programmatically-managing-focus) becomes very important.
+> ध्यान रखें, Portals के साथ काम करते वक्त [कीबोर्ड फोकस का प्रबंधन करना](/docs/accessibility.html#programmatically-managing-focus) बहुत जरूरी हो जाता है।
```
Comment (arshadkazmi42) [inline content/docs/portals.md:47]: `का प्रबंधन` => `को मैनेज`

Comment (arshadkazmi42) [inline content/docs/portals.md:47]: `Portals` => `पोर्टल्स`

File: content/docs/portals.md [hunk 3]
```diff
@@ -27,34 +27,34 @@ render() {
...
 >
-> For modal dialogs, ensure that everyone can interact with them by following the [WAI-ARIA Modal Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal).
+> मोडल डॉयलोग्स के लिए, सुनिश्चित करें कि हर कोई उनके साथ इंटरैक्ट कर सके [WAI-ARIA मोडल संलेखन प्रथाएं](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal) का पालन करते हुए।
 
-[**Try it on CodePen**](https://codepen.io/gaearon/pen/yzMaBd)
+[**CodePen पर इसे आजमाएं**](https://codepen.io/gaearon/pen/yzMaBd)
 
-## Event Bubbling Through Portals {#event-bubbling-through-portals}
+## Portals के माध्यम से इवेंट बबलिंग {#event-bubbling-through-portals}
```
Comment (arshadkazmi42) [inline content/docs/portals.md:53]: `Portals` => `पोर्टल्स`

lets change this everywhere

File: content/docs/portals.md [hunk 4]
```diff
@@ -27,34 +27,34 @@ render() {
...
 
-Even though a portal can be anywhere in the DOM tree, it behaves like a normal React child in every other way. Features like context work exactly the same regardless of whether the child is a portal, as the portal still exists in the *React tree* regardless of position in the *DOM tree*.
+एक portal DOM ट्री में कहीं भी हो, यह हर नज़रिये से एक सामान्य React चाइल्ड की तरह व्यवहार करता है। Context जैसे फीचर्स बिल्कुल उसी तरह से काम करते हैं, भले ही चाइल्ड एक portal हो, क्योंकि portal अभी भी *React ट्री* के अंदर है, इसका स्थान *DOM ट्री* में बेशक कहीं भी हो।
```
Comment (arshadkazmi42) [inline content/docs/portals.md:55]: `portal` => `पोर्टल`

File: content/docs/portals.md [hunk 5]
```diff
@@ -27,34 +27,34 @@ render() {
...
 
-This includes event bubbling. An event fired from inside a portal will propagate to ancestors in the containing *React tree*, even if those elements are not ancestors in the *DOM tree*. Assuming the following HTML structure:
+इसमें इवेंट बबलिंग भी शामिल है। एक portal के अंदर से चलाया गया इवेंट उस *React ट्री* के अनसस्टर्स तक प्रचारित करेगा, भले ही वे एलिमेंट्स *DOM ट्री* में अनसस्टर्स न हों। निम्न HTML ढांचे को मानते हुए:
```
Comment (arshadkazmi42) [inline content/docs/portals.md:57]: `portal` => `पोर्टल`

change this everywhere

File: content/docs/portals.md [hunk 6]
```diff
@@ -79,14 +80,12 @@ class Modal extends React.Component {
...
   componentDidMount() {
-    // The portal element is inserted in the DOM tree after
-    // the Modal's children are mounted, meaning that children
-    // will be mounted on a detached DOM node. If a child
-    // component requires to be attached to the DOM tree
-    // immediately when mounted, for example to measure a
-    // DOM node, or uses 'autoFocus' in a descendant, add
-    // state to Modal and only render the children when Modal
-    // is inserted in the DOM tree.
+    // Modal के चिल्ड्रन को माउंट किए जाने के बाद portal एलिमेंट को DOM ट्री
```
Comment (arshadkazmi42) [inline content/docs/portals.md:83]: `portal` => `पोर्टल`

File: content/docs/portals.md [hunk 7]
```diff
@@ -137,18 +135,18 @@ class Parent extends React.Component {
...
 }
 
 ReactDOM.render(<Parent />, appRoot);
 ```
 
-[**Try it on CodePen**](https://codepen.io/gaearon/pen/jGBWpE)
+[**CodePen पर इसे आजमाएं**](https://codepen.io/gaearon/pen/jGBWpE)
 
-Catching an event bubbling up from a portal in a parent component allows the development of more flexible abstractions that are not inherently reliant on portals. For example, if you render a `<Modal />` component, the parent can capture its events regardless of whether it's implemented using portals.
+किसी पैरेंट कौम्पोनॅन्ट के portal की इवेंट बबलिंग को पकड़ने से अधिक लचीले सार का विकास होता है जो स्वाभाविक रूप से portals पर निर्भर नहीं होते हैं। उदाहरण के लिए, यदि आप एक `<Modal />` कौम्पोनॅन्ट रेंडर करते हैं तो पैरेंट इसके इवेंट्स को पकड़ सकते हैं भले ही यह portals का उपयोग करके लागू किया गया हो।
```
Comment (arshadkazmi42) [inline content/docs/portals.md:152]: `portals` => `पोर्टल्स`

**Comment by mayankverr** [comment]
@arshadkazmi42 I wrote "portal(s)" in english because it is listed in the glossary under No translation required and also method name is `createPortal()` so I think it will make more sense, no?

**Comment by arshadkazmi42** [comment]
> @arshadkazmi42 I wrote "portal(s)" in english because it is listed in the glossary under No translation required and also method name is `createPortal()` so I think it will make more sense, no?

@mayankverr Actually you are right. I completely forgot about it. Please ignore all my comments related to `portals`

File: content/docs/portals.md (hunk 2 above)
Comment (mayankverr) [reply content/docs/portals.md]: 👍

File: content/docs/portals.md (hunk 1 above)
Comment (arshadkazmi42) [reply content/docs/portals.md:7]: not translating

File: content/docs/portals.md (hunk 2 above)
Comment (arshadkazmi42) [reply content/docs/portals.md]: not translating

File: content/docs/portals.md (hunk 3 above)
Comment (arshadkazmi42) [reply content/docs/portals.md:53]: not translating

File: content/docs/portals.md (hunk 4 above)
Comment (arshadkazmi42) [reply content/docs/portals.md:55]: not translating

File: content/docs/portals.md (hunk 5 above)
Comment (arshadkazmi42) [reply content/docs/portals.md:57]: not translating

File: content/docs/portals.md (hunk 6 above)
Comment (arshadkazmi42) [reply content/docs/portals.md:83]: not translating

File: content/docs/portals.md (hunk 7 above)
Comment (arshadkazmi42) [reply content/docs/portals.md:152]: not translating

File: content/docs/portals.md [hunk 8]
```diff
@@ -27,34 +27,34 @@ render() {
...
+  // `domNode` कोई भी एक मान्य DOM नोड है, DOM में इसका स्थान मायने नहीं रखता।
   return ReactDOM.createPortal(
     this.props.children,
     domNode
   );
 }
 ```
 
-A typical use case for portals is when a parent component has an `overflow: hidden` or `z-index` style, but you need the child to visually "break out" of its container. For example, dialogs, hovercards, and tooltips.
+Portals का एक विशिष्ट उपयोग का मामला है जब एक पैरेंट कौम्पोनॅन्ट में `overflow: hidden` या `z-index` स्टाइल होता है, लेकिन आपको चाइल्ड को नेत्रहीन उसके कंटेनर से "बाहर" करने की आवश्यकता है। जैसे डॉयलोग्स, होवरकार्ड और टूलटिप्स।
```
Comment (saranshkataria) [inline content/docs/portals.md:43]: Portals का एक विशिष्ट उपयोग का मामला है -> Portals ke एक विशिष्ट उपयोग का उदाहरण and adjust the rest of the sentence to make it grammatically  correct

Comment (saranshkataria) [reply content/docs/portals.md:43]: जैसे -> जैसे ki

File: content/docs/portals.md [hunk 9]
```diff
@@ -27,34 +27,34 @@ render() {
...
 
-> Note:
+> टिप्पणी:
 >
-> When working with portals, remember that [managing keyboard focus](/docs/accessibility.html#programmatically-managing-focus) becomes very important.
+> ध्यान रखें, Portals के साथ काम करते वक्त [कीबोर्ड फोकस को मैनेज करना](/docs/accessibility.html#programmatically-managing-focus) बहुत जरूरी हो जाता है।
 >
-> For modal dialogs, ensure that everyone can interact with them by following the [WAI-ARIA Modal Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal).
+> मोडल डॉयलोग्स के लिए, सुनिश्चित करें कि हर कोई उनके साथ इंटरैक्ट कर सके [WAI-ARIA मोडल संलेखन प्रथाएं](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal) का पालन करते हुए।
```
Comment (saranshkataria) [inline content/docs/portals.md:49]: rearrange sentence to make it more grammatically correct

File: content/docs/portals.md [hunk 10]
```diff
@@ -27,34 +27,34 @@ render() {
...
 
-## Event Bubbling Through Portals {#event-bubbling-through-portals}
+## Portals के माध्यम से इवेंट बबलिंग {#event-bubbling-through-portals}
 
-Even though a portal can be anywhere in the DOM tree, it behaves like a normal React child in every other way. Features like context work exactly the same regardless of whether the child is a portal, as the portal still exists in the *React tree* regardless of position in the *DOM tree*.
+एक portal DOM ट्री में कहीं भी हो, यह हर नज़रिये से एक सामान्य React चाइल्ड की तरह व्यवहार करता है। Context जैसे फीचर्स बिल्कुल उसी तरह से काम करते हैं, भले ही चाइल्ड एक portal हो, क्योंकि portal अभी भी *React ट्री* के अंदर है, इसका स्थान *DOM ट्री* में बेशक कहीं भी हो।
 
-This includes event bubbling. An event fired from inside a portal will propagate to ancestors in the containing *React tree*, even if those elements are not ancestors in the *DOM tree*. Assuming the following HTML structure:
+इसमें इवेंट बबलिंग भी शामिल है। एक portal के अंदर से चलाया गया इवेंट उस *React ट्री* के अनसस्टर्स तक प्रचारित करेगा, भले ही वे एलिमेंट्स *DOM ट्री* में अनसस्टर्स न हों। निम्न HTML ढांचे को मानते हुए:
```
Comment (saranshkataria) [inline content/docs/portals.md:57]: अनसस्टर्स -> spelling

File: content/docs/portals.md [hunk 11]
```diff
@@ -65,10 +65,11 @@ This includes event bubbling. An event fired from inside a portal will propagate
 </html>
 ```
 
-A `Parent` component in `#app-root` would be able to catch an uncaught, bubbling event from the sibling node `#modal-root`.
+`#app-root` में एक `पैरेंट` कौम्पोनॅन्ट अपने सिबलिंग नोड `#modal-root` से एक अज्ञात बबलिंग इवेंट को पकड़ने में सक्षम होगा।
 
-```js{28-31,42-49,53,61-63,70-71,74}
-// These two containers are siblings in the DOM
+
```
Comment (saranshkataria) [inline content/docs/portals.md:70]: why the change in numbers?

File: content/docs/portals.md [hunk 12]
```diff
@@ -137,18 +135,18 @@ class Parent extends React.Component {
...
+      <button>क्लिक</button>
     </div>
   );
 }
 
 ReactDOM.render(<Parent />, appRoot);
 ```
 
-[**Try it on CodePen**](https://codepen.io/gaearon/pen/jGBWpE)
+[**CodePen पर इसे आजमाएं**](https://codepen.io/gaearon/pen/jGBWpE)
```
Comment (saranshkataria) [inline content/docs/portals.md:150]: change to इसे CodePen पर आजमाएं

File: content/docs/portals.md (hunk 7 above)
Comment (saranshkataria) [inline content/docs/portals.md:152]: revisit first sentence once

File: content/docs/portals.md (hunk 11 above)
Comment (mayankverr) [reply content/docs/portals.md:70]: did this because after translating the comments number of lines were reduced by 2

**Comment by saranshkataria** [comment]
great work @mayankverr. Thanks! Merging this in 🥳
